packages/amplify_datastore/ios/Classes/types/model/FlutterSerializedModel.swift

```diff
@@ -149,6 +186,16 @@ struct FlutterSerializedModel: Model, JSONValueHolder {
             else if case .collection = field?.type{
                 continue
             }
+            else if case .embeddedCollection(let fieldType, _) = field?.type{
+                if case .array(let jsonArray) = value {
+                    var modifiedArray:[Any??] = []
+                    for item in jsonArray {
+                        let parsedItem = deserializeValue(value: item, fieldType: fieldType)
+                        modifiedArray.append(parsedItem)
+                    }
+                    result[key] = modifiedArray
```

Comment:
Can the `jsonArray` be null? In which case does this code add an empty array to `result[key]` instead of a null?

Reply:
if the value is `null` would this case `case .array(let jsonArray) = value` get activated? If yes, then `jsonArray` can be null

Reply:
I tested with a nullable list and passed null in the list. This case didn't get triggered.